app/models/concerns/roles.rb

```diff
@@ -8,7 +8,11 @@ module Roles
     klass.validate :roles_valid
 
     klass::ROLES.each do |role|
-      klass.scope role.pluralize.to_sym, -> { matching_role_query([role.to_s]) }
+      if role.split('_').last.numeric?
```

Comment:
**[not a blocker]** More of a tiny improvement for DRYness.

💭 Consider this instead:

```ruby
scope_name = role.split('_').last.numeric? ? role.to_sym : role.pluralize.to_sym
klass.scope scope_name, -> { matching_role_query([role.to_s]) }
```

or ultimately, if this is a rule that applies to **all strings** ending in numerics:

```ruby
# config/initializers/inflections.rb
ActiveSupport::Inflector.inflections(:en) do |inflect|
  ...
  inflect.plural /^(.*\d)$/, '\1'
end
```

 and in the latter, the code would remain unchanged.

Reply:
👍 @lostie - although i think handling the exceptions explicitly via `.irregular` is clearer, a generic numeric inflector might confuse things with patched rails magic